### ISSUE TITLE
exclude_from_localization still creates the directory with all subdirectories

### ISSUE THREAD

**Issue by musikele** [untra/polyglot#90]
Hi, 
my website supports two languages: english and italian. 
The blog is only in italian. I want to exclude blog articles from localization.

In _config.yml, I have inserted: 
```
exclude_from_localization: ["assets", "css", "blog"]
```
(note: after compilation, my posts end in `/blog` directory, for example: `/blog/2018/04/26/title/index.html`).

After generating the website, I get: 

- /blog/index.html is correctly generated 
- /en/blog/ does not contain any index.html, BUT 
- /en/blog/2018/04/03/.... exists and contains the articles. 

How can I prevent the plugin from generating anything contained in the `blog` directory ?

**Comment by musikele** [comment]
excluding a single article, like this, won't work: 

```
exclude_from_localization: ["assets", "css", "blog", "blog//2018//04//23//article-folder-n1"]
```

**Comment by untra** [comment]
Hello @musikele 👋
Can you provide a link to your repo? I'll try to clone it down and take a look.

Is it a problem those en blog posts are being generated? Could you just rm the en/blog/ directory after building, but before uploading up the generated site?

**Comment by musikele** [comment]
Hi, thanks for your quick response. 

Unfortunately I cannot give you the link to the repo since it is a private product. 

I might recreate the issue with the minimum necessary data to see it. 

> Is it a problem those en blog posts are being generated?  

At first yes; the SEO department said these urls are being found by search engines and clicked by users. 
My solution has been to add a rule in robots.txt to not track those files. 

> Could you just rm the en/blog/ directory after building, but before uploading up the generated site?

The process of building the website is very complex, involving build steps, checkout from git tu svn, etc. Adding a "remove" step in the build process might require people granting authorizations, security reviews, months passing ... :(

**Comment by untra** [comment]
Well, the polyglot plugin is designed to generate equivalent pages and
posts in each language. It does this to encourage making sites that are
100% available in each language. This is a feature, not a bug,
unfortunately 😬

Another approach (I would not recommended this) is to actually create those
English blog posts with no content in them. That way those pages are
generated with no content. That can get problematic if you have a lot of
Italian blog posts though.

I would say the easiest approach is to just `rm en/blog/` directory, if you
don't want those pages to exist at all. That can be done in exactly one
line in a shell script for you. I'm sorry you have a complex bureaucracy to
navigate. My suggestion is to make that change anyways, and ask for
forgiveness instead of permission.

Cheers

On Jul 9, 2018 7:44 AM, "Michele Nasti" <notifications@github.com> wrote:

Hi, thanks for your quick response.

Unfortunately I cannot give you the link to the repo since it is a private
product.

I might recreate the issue with the minimum necessary data to see it.

Is it a problem those en blog posts are being generated?

At first yes; the SEO department said these urls are being found by search
engines and clicked by users.
My solution has been to add a rule in robots.txt to not track those files.

Could you just rm the en/blog/ directory after building, but before
uploading up the generated site?

The process of building the website is very complex, involving build steps,
checkout from git tu svn, etc. Adding a "remove" step in the build process
might require people granting authorizations, security reviews, months
passing ... :(

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/untra/polyglot/issues/90#issuecomment-403483949>, or mute
the thread

**Comment by musikele** [comment]
Hi there, 
Thanks for your feedback. I'll evaluate the "one line" approach, but just for future people searching for the same feature, here is another option based on your comment : 
- create a layout called "redirect.html". Put some javascript that redirects users to other pages based on a variable `{{page.redirect_url}}`
- create fake articles in the other language and use the layout `redirect`
- redirect users to the home page (in my case) by using a `redirect_url: ...` variable in the front matter. 

I too encourage to `rm` the directory though, that's obviously the simplest and best way to do this.

**Comment by vlsi** [comment]
>create fake articles in the other language and use the layout redirect

That looks like something boring.

>create fake articles in the other language and use the layout redirect

That is a key point. Do you expect "language selection link" to just disappear when displaying italian blogpost?
Do you expect "language selection link" to be present? If so, where should it point to?

I think this issue and #63 are related.

**Comment by musikele** [comment]
The blog section exists only in Italian and when we switch to English the blog link in the navbar disappears. I don't know how but search engines have found these articles and started indexing them with the wrong language tag, with impacts on seo. At the end I had to tweak robots.txt and, for some time, I cannot delete these english articles but keep them and redirect to home page.

**Comment by musikele** [comment]
For me the issue can be considered closed; #63 is the feature request that maps 100% my needs.
